python/lsst/dax/imgserv/cli_v1.py

```diff
@@ -0,0 +1,176 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+
+'''
+This module implements the command line interface (CLI) to ImageServ.
+
+@author: Brian Van Klaveren, SLAC
+@author: Kenny Lo, SLAC
+
+'''
+
+import os
+import json
+
+import click
+
+import lsst.log as log
+import lsst.daf.persistence as dafPersist
+
+from .locateImage import (
+        image_open_v1, W13DeepCoaddDb, W13RawDb, W13CalexpDb
+)
+
+from .image import Image
+from .dispatch_v1 import Dispatcher
+from .hashutil import Hasher
+from .jsonutil import flatten_json
+
+ROOT = os.path.abspath(os.path.dirname(__file__))
+
+@click.command()
+@click.argument("--config")
+@click.argument("--in_req", type=click.Path(exists=True))
+@click.argument("--out_dir", type=click.Path(exists=True))
+def exec_command(config_dir, in_req, out_dir):
+    """ Command Line Interface: Process query to return image file
+        in output directory.
+    """
+    cli = ImageServCLI(config_dir, in_req, out_dir)
+    cli.process_request()
+
+class ImageServCLI(object):
+    """ Module to implement CLI for ImageServ.
+
+    """
+    def __init__(self, config_dir, in_req, out_dir):
+        # load the configuration file
+        if config_dir is not None:
+            config = os.path.join(config_dir, "settings.json")
+        else:
+            config_dir = os.path.join(ROOT, "config")
+            config = os.path.join(config_dir, "settings.json")
+        with open(config) as f:
+            self._config = json.load(f)
+            # strip /tests from dataroot if cur_dir is /tests
+            cwd = os.getcwd()
+            f.close()
+         # configure the log file (log4cxx)
+        log.configure(os.path.join(config_dir, "log.properties"))
+        self._in_req = in_req
+        self._out_dir = out_dir
+        self._dispatcher = Dispatcher(config_dir)
+
+    def process_request(self):
+        # req: JSON object 
+        errors, req = self._parse_req()
+        w13db = self._get_ds(req)
+        if w13db:
+            img_getter = image_open_v1(w13db, self._config)
+            result = self.dispatch(img_getter, req)
+            name = req["name"]
+            fp_image = self._save_result(result, name)
+            if self._check_result(req, fp_image):
+                print(name+": PASS")
+            else:
+                print(name+": FAIL")
+        else:
+            raise Exception("ImageServCLI", "w13db should NOT be None")
+
+    def dispatch(self, img_getter, req):
+        # direct the request to the best fit image API on parameters
+        req_params = self._get_params(req)
+        api = self._dispatcher.find_api(req_params)
+        # call the API with the params
+        try:
+            result = api(img_getter, req_params)
+            return result
+        except:
```

Comment:
Specify exception here (also PEP8), use `except Exception` if you must.

Reply:
Exception to be captured and handled higher up.